lightningd/log.c

```diff
@@ -564,27 +565,39 @@ void logv(struct logger *log, enum log_level level,
 {
 	int save_errno = errno;
 	struct log_entry *l = new_log_entry(log, level, node_id);
+	char *log_msg = NULL;
 
 	/* This is WARN_UNUSED_RESULT, because everyone should somehow deal
 	 * with OOM, even though nobody does. */
-	if (vasprintf(&l->log, fmt, ap) == -1)
+	if (vasprintf(&log_msg, fmt, ap) == -1)
 		abort();
 
-	size_t log_len = strlen(l->log);
+	char **lines = tal_strsplit(
+	    log, json_escape_unescape(log, (struct json_escape *)log_msg), "\n",
+	    STR_NO_EMPTY);
```

Comment:
In 2019 we changed l->log from a tal pointer to a raw pointer (see how we free() it in delete_entry()), to save memory.  It's a fair optimization.

I suggest:

1. If the string doesn't contain "\" at all, fast path: l->log = log_msg;
2. BTW, unescape can reutrn NULL on a bad escape.  So handle that case too (like the above).
3. Otherwise, copy each one using strdup() instead of tal_strdup()